setup.py

```diff
@@ -113,7 +113,7 @@
     # Minimum 8.0.0 to be able to use .to_reader()
     "pyarrow>=8.0.0",
     # For smart caching dataset processing
-    "dill>=0.3.0,<0.3.7",  # tmp pin until next 0.3.7 release: see https://github.com/huggingface/datasets/pull/5166
+    "dill>=0.3.0",
```

Comment:
I think we should pin <0.3.8 no ? In case a future version of dill breaks the deterministic behaviors the cache is relying on

Reply:
especially since determinism is not officially supported yet: https://github.com/uqfoundation/dill/pull/501